Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```diff
@@ -126,6 +131,30 @@ export class LGServer {
     this.connection.listen();
   }
 
+  protected definitionHandler(params: TextDocumentPositionParams): Location | undefined {
+    const document = this.documents.get(params.textDocument.uri);
+    if (!document) {
+      return;
+    }
+    const wordRange = getRangeAtPosition(document, params.position);
+    const word = document.getText(wordRange);
+    const curFileResult = this._lgFile?.templates.find((t) => t.name === word);
+
+    if (curFileResult?.range) {
+      return Location.create(
+        params.textDocument.uri,
+        Range.create(curFileResult.range.start.line - 1, 0, curFileResult.range.end.line, 0)
+      );
+    }
+
+    const refResult = this._templateDefinitions[word];
+    if (refResult) {
+      this.connection.sendNotification('GotoDefinition', refResult);
```

Comment:
FYI: doing it this way is not gonna play nice with cmd+click (windows: ctrl+click) for go to defintion.

Reply:
Yeah, I agree. Current solution is some kind trade off for our model managing LG and LU files. 
I think we could use the latest version of monaco-editor/react in the future. It supports multi-model editor.
![image](https://user-images.githubusercontent.com/17074777/111403724-7194d700-8708-11eb-9337-c3faf3141e62.png)

Reply:
cool